File: app/helpers/analyzer.rb
module Analyzer

  # find characters that are in the final word
  # and characters that are not in the final word
  def analyze_chars(words)
    all_chars = ('a'..'z').to_a
    chars_in = words.reduce(all_chars){ |chars,word| chars & word.chars }
    chars_out = all_chars - words.reduce([]){ |chars,word| chars | word.chars }
    { found: chars_in, eliminated: chars_out }
  end

end

File: app/javascript/packs/core/analyzer.js
import { intersect, union, subtract } from "./utils"

const alphebet = 'abcdefghijklmnopqrstuvwxyz'.split('')

export default function analyzer(words) {
    let found = words.reduce((chars, word) =>
        intersect(chars, word), alphebet)

    let eliminated = subtract(alphebet,
        words.reduce((chars, word) => union(chars, word), []))

    return {
        found,
        eliminated
    }
}